### PR TITLE
Added: GZip compression feature (#621)

### LINKED ISSUE #621: API Feature - Implement GZIP for POST/PUT
The engine is introducing a new feature allowing to optionally compress request
payloads (ie.  POST/PUT body, http or https).

To test it:

Perform a `saveObject` with the content encoded with GZIP
Don't forget to add: `"Content-Encoding: gzip"` as a header.

We will discuss a specification for a more detailed implementation.
Like adding a `enableGZIP` flag in client's configuration.

### PR DESCRIPTION
**Pull request by Ant-hem**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no     
| Related Issue     | Fix #621
| Need Doc update   | yes

### CODE REVIEW

**Comment by Ant-hem** [comment]
> Perhaps we had the discussion just after this PR but could we change this implementation for the `enum` version of it? Not a `GzipEnabled` boolean but rather a `Compression.{None,GZIP}` enum-like one?

Yes definitely! I did it for Java, will do for C# now :) This proposition was the first draft.

**Comment by Ant-hem** [comment]
@aseure updated with 277f880.

File: src/Algolia.Search/Clients/SearchConfig.cs [hunk 1]
```diff
@@ -88,6 +88,11 @@ public SearchConfig(string applicationId, string apiKey) : base(applicationId, a
             hosts.AddRange(commonHosts);
 
             DefaultHosts = hosts;
+
+            Compression = CompressionType.GZIP;
```
Comment (aseure) [inline src/Algolia.Search/Clients/SearchConfig.cs]: If it's the default value, don't we want to change it for `NONE`?

Comment (Ant-hem) [reply src/Algolia.Search/Clients/SearchConfig.cs]: 👍 Just updated it.

**Pull request merged**